Source/Login/Backend/providers/getter.go

```diff
@@ -4,15 +4,34 @@ type Getter interface {
 	GetProviderByID(id string) (Provider, error)
 }
 
-func NewGetter() Getter {
-	return &getter{}
+func NewGetter(configuration Configuration) Getter {
+	return &getter{
+		configuration,
+	}
 }
 
-type getter struct{}
+type getter struct {
+	configuration Configuration
+}
 
 func (g *getter) GetProviderByID(id string) (Provider, error) {
+	provider, ok := g.configuration.Providers()[id]
+	if !ok {
+		return Provider{
+			ID:       id,
+			Display:  id + " Display Name",
```

Comment:
This was just for testing purposes, we can just get rid of it - it looks bad...
```suggestion
			Display:  id,
```